Transaction is not seen as reverted with Anvil as opposed to hardhat EVM

### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (00854b6 2024-06-06T00:17:50.550156000Z)

### What command(s) is the bug in?

hardhat test --network localhost test/PoolGeneral.js

### Operating System

Linux

### Describe the bug

I'm trying to figure out why a transaction that should get reverted can't be confirmed to be reverted when I test it against Anvil (It does get reverted as expected using only hardhat EVM though). So I wrote this little function and try to ensure it's reverted to keep it simple
```
function withdraw(uint256 amount) external whenNotPaused {
        require(0 > 1, "0 is not greater than 1");
```
and my test is the following. I expect it to fail and say the exception is different from what I expect but at least it will be reverted as you can see when I run it using hardhat EVM
```
it("Stake for this account does not exist", async () => {
        await expect(
            MySimpleContract.connect(user1).withdraw(ethers.constants.MaxUint256)
        ).to.be.revertedWith("stake for this account does not exist");
    })
```
```
> hardhat test test/PoolGeneral.js (hardhat EVM)

  0 passing (520ms)
  1 failing

  1) End-to-end main pool general case testing
       Stake for this account does not exist:
     AssertionError: Expected transaction to be reverted with stake for this account does not exist, but other exception was thrown: Error: VM Exception while processing transaction: **reverted** with reason string '0 is not greater than 1'
```

but if I run against Anvil then it doesn't fail

```
> hardhat test --network localhost test/PoolGeneral.js

  0 passing (321ms)
  1 failing

  1) End-to-end main pool general case testing
       Stake for this account does not exist:

      AssertionError: Expected transaction to be reverted
      + expected - actual

      -**Transaction NOT reverted**.
      +Transaction reverted.
```

I then tried to check what the receipt says. If I check the corresponding transaction receipt (using `cast re 0xfefc29fb4a91b306e4c4505a9cb6e743490f376eabfa81c54936acbf203cb548` in Anvil I don't find any
```
{
  hash: '0xfefc29fb4a91b306e4c4505a9cb6e743490f376eabfa81c54936acbf203cb548',
  type: 2,
  accessList: [],
  blockHash: null,
  blockNumber: null,
  transactionIndex: null,
  confirmations: 0,
  from: '0x3C44CdDdB6a900fa2b585dd299e03d12FA4293BC',
  gasPrice: BigNumber { _hex: '0x5b2052f9', _isBigNumber: true },
  maxPriorityFeePerGas: BigNumber { _hex: '0x3b9aca00', _isBigNumber: true },
  maxFeePerGas: BigNumber { _hex: '0x5b2052f9', _isBigNumber: true },
  gasLimit: BigNumber { _hex: '0x3b8be358', _isBigNumber: true },
  to: '0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9',
  value: BigNumber { _hex: '0x00', _isBigNumber: true },
  nonce: 0,
  data: '0x2e1a7d4dffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff',
  r: '0xf1e3fdb7787c7f935ada002dc6018f07d37bf00bf7d346d77608fd969a558efd',
  s: '0x57113a52eb77ef1c9f4187eb2cd4aa083622ad98c03d9ac688712b8d154885a9',
  v: 0,
  creates: null,
  chainId: 31337,
  wait: [Function (anonymous)]
}
```
and I used this code to this output

```
const provider = ethers.provider
        try {
            tx = await MySimpleContract.connect(user1).withdraw(ethers.constants.MaxUint256)
            console.log(tx)
            const receipt = await provider.getTransactionReceipt(tx.hash)
            console.log("Successful transaction receipt:", receipt)
        } catch (error) {
            tx = error.tx
            console.log(tx)
            const receipt = await provider.getTransactionReceipt(tx.hash)
            console.log("Failing transaction receipt:", receipt)        }
    });
```

if I use ganache instead of Anvil I then get a receipt (but for some reason the transaction is still not seen as reverted)
```
{
  hash: '0xbf93fec4cce3344cf6fd1ef16a87d7a7ec850e503936433793323a27a38589b7',
  type: 2,
  accessList: [],
  blockHash: '0x7f32d98d4954252011b2cc04470d9bbe80e741d470435b97948918e91c5c8b4e',
  blockNumber: 8,
  transactionIndex: 0,
  confirmations: 1,
  from: '0x1A55247bFBeadb633eF96445D67056bBFDc0f99f',
  gasPrice: BigNumber { _hex: '0x19f89429', _isBigNumber: true },
  maxPriorityFeePerGas: BigNumber { _hex: '0x063ab194', _isBigNumber: true },
  maxFeePerGas: BigNumber { _hex: '0x19f89429', _isBigNumber: true },
  gasLimit: BigNumber { _hex: '0x3b8be358', _isBigNumber: true },
  to: '0x65FC103eb11f91650BE7e2B23d4cc4BBB913B979',
  value: BigNumber { _hex: '0x00', _isBigNumber: true },
  nonce: 0,
  data: '0x2e1a7d4dffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff',
  r: '0x8b1882f742954fdecd7928987cbb99bca82c692ce505e66e95b9a69edefe698d',
  s: '0x4a04c49f4a5ddb5d7f2ab955e824939e02c1f22bafd3a1d5f382440b356d1208',
  v: 1,
  creates: null,
  chainId: 1337,
  wait: [Function (anonymous)]
}

# Why does it say Successful here ?? It doesn't make any sense
Successful transaction receipt: {
  to: '0x65FC103eb11f91650BE7e2B23d4cc4BBB913B979',
  from: '0x1A55247bFBeadb633eF96445D67056bBFDc0f99f',
  contractAddress: null,
  transactionIndex: 0,
  gasUsed: BigNumber { _hex: '0x5e19', _isBigNumber: true },
  logsBloom: '0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
  blockHash: '0x7f32d98d4954252011b2cc04470d9bbe80e741d470435b97948918e91c5c8b4e',
  transactionHash: '0xbf93fec4cce3344cf6fd1ef16a87d7a7ec850e503936433793323a27a38589b7',
  logs: [],
  blockNumber: 8,
  confirmations: 1,
  cumulativeGasUsed: BigNumber { _hex: '0x5e19', _isBigNumber: true },
  effectiveGasPrice: BigNumber { _hex: '0x19f89429', _isBigNumber: true },
  status: 0,
  type: 2,
  byzantium: true
}
```
But there is the same problem regarding the fact **it thinks the tx succeeded** (you can see the **status is 0 = failed**). So something goes wrong here with a slight difference between ganache and Anvil but still they both fail.

--- a/crates/anvil/src/eth/backend/executor.rs
+++ b/crates/anvil/src/eth/backend/executor.rs
@@ -270,9 +270,9 @@ impl<'a, 'b, DB: Db + ?Sized, Validator: TransactionValidator> Iterator
         };
         let env = self.env_for(&transaction.pending_transaction);
 
-        // check that we comply with the block's gas limit
+        // check that we comply with the block's gas limit, if not disabled
         let max_gas = self.gas_used.saturating_add(env.tx.gas_limit as u128);
-        if max_gas > env.block.gas_limit.to::<u128>() {
+        if !env.cfg.disable_block_gas_limit && max_gas > env.block.gas_limit.to::<u128>() {
             return Some(TransactionExecutionOutcome::Exhausted(transaction))
         }
 

--- a/crates/anvil/tests/it/transaction.rs
+++ b/crates/anvil/tests/it/transaction.rs
@@ -227,6 +227,28 @@ async fn can_reject_too_high_gas_limits() {
     let _ = pending.unwrap();
 }
 
+// <https://github.com/foundry-rs/foundry/issues/8094>
+#[tokio::test(flavor = "multi_thread")]
+async fn can_mine_large_gas_limit() {
+    let (api, handle) = spawn(NodeConfig::test().disable_block_gas_limit(true)).await;
+    let provider = handle.http_provider();
+
+    let accounts = handle.dev_wallets().collect::<Vec<_>>();
+    let from = accounts[0].address();
+    let to = accounts[1].address();
+
+    let gas_limit = api.gas_limit().to::<u128>();
+    let amount = handle.genesis_balance().checked_div(U256::from(3u64)).unwrap();
+
+    let tx =
+        TransactionRequest::default().to(to).value(amount).from(from).with_gas_limit(gas_limit * 3);
+
+    // send transaction with higher gas limit
+    let pending = provider.send_transaction(WithOtherFields::new(tx)).await.unwrap();
+
+    let _resp = pending.get_receipt().await.unwrap();
+}
+
 #[tokio::test(flavor = "multi_thread")]
 async fn can_reject_underpriced_replacement() {
     let (api, handle) = spawn(NodeConfig::test()).await;

